libraries/chain/webassembly/runtimes/eos-vm.cpp

```diff
@@ -301,7 +322,17 @@ std::unique_ptr<wasm_instantiated_module_interface> eos_vm_runtime<Impl>::instan
       else
 #endif
          bkend = std::make_unique<backend_t>(code, code_size, nullptr, options, false, false); // false, false <--> 2-passes parsing, backend does not own execution context (execution context is reused per thread)
+                                                                                               //
       eos_vm_host_functions_t::resolve(bkend->get_module());
+
+      // check sync_call entry point
+      sync_call_supported = false;
+      uint32_t sync_call_idx = bkend->get_module().get_exported_function("sync_call");
+      if (sync_call_idx < std::numeric_limits<uint32_t>::max()) {
+         // signature has already been checked when the contract was deployed
+         sync_call_supported = true;
+      }
+
```

Comment:
Instead of caching `sync_call_supported`, I think I should move this code to `do_sync_call`. it is cleaner and much fewer files are touched.

Reply:
Much cleaner. https://github.com/AntelopeIO/spring/pull/1266/commits/e94850e022228c7211133b0eb9cbed7e92f7c1a3
https://github.com/AntelopeIO/spring/pull/1266/commits/ebce5e2913b96b209cab732992fc31125c132b08